### PR TITLE
fix: defer handling v5 flags to core

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

fix

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Not yet

**Summary**
- We're handing devtool flag in our own way, core already exports it and hence should be supported by core and not by cli side.

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -210,15 +210,6 @@ const builtInFlags = [
...
     },
-    {
-        name: 'devtool',
-        usage: '--devtool <value>',
-        type: String,
-        negative: true,
-        alias: 'd',
-        description: 'Determine source maps to use',
-        link: 'https://webpack.js.org/configuration/devtool/#devtool',
-    },
```
Comment (snitin315) [inline packages/webpack-cli/lib/utils/cli-flags.js:221]: will `webpack-4` still support `webpack --devtool` ?

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js:221]: seems like no then we should add it conditionally

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js:221]: added not to drop it once webpack@4 is no longer supported

**Review by alexander-akait — COMMENTED** [review]
We should use union object here, it is reduce quality of code

**Review by alexander-akait — CHANGES_REQUESTED** [review]
Also it should be done for all arguments from core and built-in, not only for devtool

**Comment by anshumanv** [comment]
yep rest wip

**Comment by anshumanv** [comment]
some problems with target - core is not merging it correctly, will take a look at it later

**Comment by anshumanv** [comment]
Added a note to move it later

**Review by alexander-akait — APPROVED** [review]
Looks good

**Comment by alexander-akait** [comment]
Need fir tests

**Comment by anshumanv** [comment]
yep something strange, lemme see

File: packages/webpack-cli/lib/webpack-cli.js [hunk 2]
```diff
@@ -328,19 +315,27 @@ class WebpackCLI {
...
                 options.mode = process.env.NODE_ENV;
             }
-
-            if (args.name) {
-                options.name = args.name;
-            }
-
-            if (typeof args.stats !== 'undefined') {
-                options.stats = args.stats;
+            if (args.entry) {
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/webpack-cli.js:318]: Why it here?

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 3]
```diff
@@ -269,12 +275,13 @@ const coreFlags = cli
           }
 
           const inBuiltIn = builtInFlags.find((builtInFlag) => builtInFlag.name === flag);
+          const legacyMetadata = flagsForWebpack4[flag] || {};
 
           if (inBuiltIn) {
-              return { ...meta, name: flag, group: 'core', ...inBuiltIn };
+              return { ...meta, name: flag, group: 'core', ...inBuiltIn, ...legacyMetadata };
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/utils/cli-flags.js:281]: I don't think it is right, why we need double `...`?

File: packages/webpack-cli/lib/webpack-cli.js (hunk 2 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:318]: some bugs in cli.processArguments, it's not able to handle multiple flags, will check later

Comment (alexander-akait) [reply packages/webpack-cli/lib/webpack-cli.js:318]: We need to fix it here

File: packages/webpack-cli/lib/utils/cli-flags.js (hunk 3 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js]: yep redundant, removed

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 4]
```diff
@@ -105,10 +105,10 @@ const builtInFlags = [
     },
     {
         name: 'merge',
-        usage: '--merge',
+        usage: '--config <first-config> --config <second-config> --merge',
         alias: 'm',
         type: Boolean,
-        description: 'Merge two or more configurations using webpack-merge e.g. -c ./webpack.config.js -c ./webpack.test.config.js --merge',
+        description: 'Merge two or more configurations using webpack-merge',
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/utils/cli-flags.js:111]: Can we do it in separate PR?

Comment (snitin315) [reply packages/webpack-cli/lib/utils/cli-flags.js:111]: It's already done. need rebase I think.

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js:111]: yep need rebase, in near future
